### PR TITLE
Minimal codegen for tfhe_rust dialect

### PR DESCRIPTION
**Pull request by j2kun**
Blocked by https://github.com/google/heir/pull/217

Outline of the process:

- Write an analysis pass to generate variable names for each SSA value
- Code gen minimal ops to support composing two encrypted bits into a bitmask, and applying a lookup table.

Next:
- Op to create a lookup table from a static constant
- Re-use `scf.forall` op to represent the parallel-executed regions of the circuit (pass that precedes this codegen step will handle toposorting the circuit)
- Figure out how we're gonna do e2e testing of this.

It seems like an e2e test that compiles the result with rustc and tfhe-rs is not suitable for living in this project. But we should have some e2e test flow, even if manual, that asserts the generated code works against a specific tfhe-rs API version.

### CODE REVIEW

File: lib/Analysis/SelectVariableNames/SelectVariableNames.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#include "include/Analysis/SelectVariableNames/SelectVariableNames.h"
+
+#include <iostream>
```
Comment (asraa) [inline lib/Analysis/SelectVariableNames/SelectVariableNames.cpp:3]: nit: i always keep this in after debugging by accident, do you need this?

File: lib/Target/TfheRust/TfheRustEmitter.cpp [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    if (failed(translate(op))) {
+      return failure();
+    }
+  }
+
+  return success();
+}
+
+LogicalResult TfheRustEmitter::printOperation(func::FuncOp funcOp) {
+  os << "pub fn " << funcOp.getName() << "(\n";
```
Comment (asraa) [inline lib/Target/TfheRust/TfheRustEmitter.cpp:105]: kinda silly, but should the func private visibility attribute be used to determine whether the rust fn is `pub`?

**Review by asraa — APPROVED** [review]
Nice!

File: lib/Target/TfheRust/TfheRustEmitter.cpp (hunk 2 above)
Comment (j2kun) [reply lib/Target/TfheRust/TfheRustEmitter.cpp:105]: I suppose we could.... I'm not entirely sure what that would do for us though. I'll think about it and add it to my TODO list.

File: lib/Analysis/SelectVariableNames/SelectVariableNames.cpp (hunk 1 above)
Comment (j2kun) [reply lib/Analysis/SelectVariableNames/SelectVariableNames.cpp]: Done, thanks!

**Pull request merged**